File: epi_judge_python_solutions/reverse_list.py
from list_node import ListNode
from test_framework import generic_test


def reverse_list(head: ListNode) -> ListNode:
    dummy = ListNode(0)
    while head:
        dummy.next, head.next, head = head, dummy.next, head.next
    return dummy.next


if __name__ == '__main__':
    exit(
        generic_test.generic_test_main('reverse_list.py', 'reverse_list.tsv',
                                       reverse_list))

File: epi_judge_java_solutions/epi/ReverseList.java

package epi;

import epi.test_framework.EpiTest;
import epi.test_framework.GenericTest;

public class ReverseList {
  @EpiTest(testDataFile = "reverse_list.tsv")

  public static ListNode<Integer> reverseList(ListNode<Integer> head) {
    ListNode<Integer> prev = null, curr = head;
    while (curr != null) {
      ListNode<Integer> temp = curr.next;
      curr.next = prev;
      prev = curr;
      curr = temp;
    }
    return prev;
  }

  public static void main(String[] args) {
    System.exit(
        GenericTest
            .runFromAnnotations(args, "ReverseList.java",
                                new Object() {}.getClass().getEnclosingClass())
            .ordinal());
  }
}
